Servers/UI/OJS.Servers.Ui/ClientApp/src/components/profile/profile-submissions/ProfileSubmisssions.tsx

```diff
@@ -1,227 +1,94 @@
-        ),
-        [],
-    );
-
-    const currentSubmissions = useMemo(
-        () => isEmpty(selectValue.key)
-            ? userSubmissions
-            : userByContestSubmissions,
-        [ userByContestSubmissions, userSubmissions, selectValue ],
-    );
-
-    const handleClearFiltersClick = () => {
-        setSelectValue(defaultState.state.selectValue);
-        clearPageValue();
-        setSubmissionsPage(defaultState.state.initialPage);
-        initiateUserSubmissionsForProfileQuery(getDecodedUsernameFromProfile(), defaultState.state.initialPage);
-    };
-
-    const renderSubmissionsList = useCallback(
-        () => {
-            if (userSubmissionsLoading || userSubmissionsByContestLoading) {
-                return (
-                    <div style={{ ...flexCenterObjectStyles, marginTop: '10px' }}>
-                        <SpinningLoader />
-                    </div>
-                );
-            }
-
-            if (currentSubmissions.length === 0) {
-                return (
-                    <div className={styles.noSubmissionsFound}>
-                        No submissions found.
-                    </div>
-                );
-            }
-
-            return (
-                <>
-                    <PaginationControls
-                        count={pagesInfo.pagesCount}
-                        page={submissionsPage}
-                        onChange={handlePageChange}
-                    />
-                    <List
-                        values={currentSubmissions}
-                        itemFunc={renderSubmissionRow}
-                        itemClassName={styles.submissionRow}
-                    />
-                </>
-            );
-        },
-        [
-            currentSubmissions,
-            userSubmissions,
-            userByContestSubmissions,
-            userSubmissionsLoading,
-            renderSubmissionRow,
-        ],
-    );
+    }, [ dispatch, userSubmissions ]);
 
     return (
-        <>
-            <Heading type={HeadingType.primary}>Submissions:</Heading>
-            <div className={styles.filtersContainer}>
-                {renderSubmissionsDropdown()}
-                <div className={styles.clearFiltersBtn}>
-                    <Button
-                        type={ButtonType.secondary}
-                        onClick={() => handleClearFiltersClick()}
-                        className={styles.button}
-                        text="clear contest filter"
-                        size={ButtonSize.small}
-                    />
+        areSubmissionsLoading
+            ? (
+                <div style={{ ...flexCenterObjectStyles, marginTop: '10px' }}>
+                    <SpinningLoader />
                 </div>
-            </div>
-                {renderSubmissionsList()}
-        </>
+            )
+            : shouldRender
+                ? (
+                    <SubmissionsGrid
+                      isDataLoaded={!areSubmissionsLoading}
```

Comment:
doesnt sibmissionsgrid has loader based on the isDataLoadding property here, which would make the check above redundant, and also in this way do we return message if there are no submission available? I think we can simplify this conditional rendering logic

Reply:
SubmissionsGrid does include a loader, so i've removed this check here